src/main/java/org/apache/maven/plugins/dependency/analyze/AnalyzeDuplicateMojo.java

```diff
@@ -151,15 +150,9 @@ private void createMessage( Set<String> duplicateDependencies, StringBuilder sb,
 
     private Set<String> findDuplicateDependencies( List<Dependency> modelDependencies )
     {
-        List<String> modelDependencies2 = new ArrayList<>();
-        for ( Dependency dep : modelDependencies )
-        {
-            modelDependencies2.add( dep.getManagementKey() );
-        }
-
-        // @formatter:off
+        List<String> modelDependencies2 =
+                modelDependencies.stream().map( Dependency::getManagementKey ).collect( Collectors.toList() );
         return new LinkedHashSet<>(
-                CollectionUtils.disjunction( modelDependencies2, new LinkedHashSet<>( modelDependencies2 ) ) );
-        // @formatter:on
+                Util.symmetricDifference( modelDependencies2, new LinkedHashSet<>( modelDependencies2 ) ) );
```

Comment:
I struggled on that one.
Given the two sets are the same, I think the result will always be an empty set.
So I would just simply remove that call.

Reply:
No they are not the same. The first list (modelDependencies2) contains two entries (junit) while the hashset contains only a single entry of that and the result must be the single junit-jar difference. Check the unit test: `TestAnalyzeDuplicateMojo` testcase `testDuplicate` and `testDuplicate2` which references `src/test/resources/unit/duplicate-dependencies/plugin-config.xml` and the `src/test/resources/unit/duplicate-dependencies/plugin-config2.xml` which contains exactly those cases.

Reply:
Ah, got it.  But then, wouldn't it be simpler and more understandable :
```
    /**
     * Remove exactly one instance of each element from the give list. 
     */
    static <O> List<O> symmetricDifference( Collection<O> elements )
    {
        List<O> list = new ArrayList<>( elements );
        Set<O> set = new HashSet<>( elements );
        list.removeIf( set::remove );
        return list;
    }
```

Reply:
It could even be inlined with a simple comment...

Reply:
The shortest form could be:
```
    private Set<String> findDuplicateDependencies( List<Dependency> modelDependencies )
    {
        List<String> modelDependencies2 = modelDependencies.stream()
                .map( Dependency::getManagementKey ).collect( Collectors.toList() );
        // remove one instance of each element from the list
        modelDependencies2.removeIf( new HashSet<>( modelDependencies2 )::remove );
       // keep a single instance of each duplicate
        return new LinkedHashSet<>( modelDependencies2 );
    }
```